src/main/java/Duke.java

```diff
@@ -1,10 +1,166 @@
+import jdk.jfr.Event;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+
 public class Duke {
-    public static void main(String[] args) {
+    public static void loadTasksFromFile(ArrayList<Task> tasks) throws IOException, DukeException {
+        String directoryPath = "./data";
+        String filePath = directoryPath.concat("/duke.txt");
+
+        File directory = new File(directoryPath);
+        File dataFile = new File(filePath);
+
+        if (!directory.exists()) {
+            throw new DukeException("OOPS!! Directory does not exist");
+        } else if (!dataFile.exists()) {
+            dataFile.createNewFile();
+        }
+
+        Scanner f = new Scanner(dataFile);
+
+        while (f.hasNext()) {
+            String[] task = f.nextLine().split("|");
+            String taskType = task[0];
+            boolean isDone = (Integer.parseInt(task[1]) == 1);
+            String description = task[2];
+            String time = task.length == 4 ? task[3] : null;
+
+            if (taskType.equals("T")) {
+                tasks.add(new TodoTask(description, isDone));
+            } else if (taskType.equals("D")) {
+                tasks.add(new DeadlineTask(description, isDone, time));
+            } else {
+                tasks.add(new EventTask(description, isDone, time));
+            }
+        }
+    }
+
+    public static void saveTasksToFile(ArrayList<Task> tasks) throws IOException {
+        FileWriter fw = new FileWriter("./data/duke.txt");
```

Comment:
Really nice use of file writer btw! just one suggestion you can check `try-with-resources`